plugins/woocommerce/tests/performance/setup/add-customer-order.js

```diff
@@ -0,0 +1,52 @@
+/* eslint-disable jsdoc/require-property-description */
+/* eslint-disable @woocommerce/dependency-group */
+/* eslint-disable import/no-unresolved */
+/**
+ * k6 dependencies
+ */
+import encoding from 'k6/encoding';
+import http from 'k6/http';
+import { findBetween } from 'https://jslib.k6.io/k6-utils/1.2.0/index.js';
+
+/**
+ * Internal dependencies
+ */
+import { base_url, admin_username, admin_password } from '../config.js';
+
+/**
+ * Convert Cart & Checkout pages to shortcode.
+ */
+export function addCustomerOrder() {
+	function addOrderToCustomer() {
+		let response;
+
+		const credentials = `${ admin_username }:${ admin_password }`;
+		const encodedCredentials = encoding.b64encode( credentials );
+		const requestHeaders = {
+			Authorization: `Basic ${ encodedCredentials }`,
+			'Content-Type': 'application/json',
+		};
+
+		response = http.get( `${ base_url }/wp-json/wc/v3/customers`, {
+			headers: requestHeaders,
+			tags: { name: 'API - Retrieve Customer' },
+		} );
+
+		const customerId = findBetween( response.body, '"id":', ',' );
```

Comment:
@adimoldovan I updated the code in add-customer-order.js completely to align with this. However, found some obstacles in the design:

1. The test suite has `customer_username` stored in `config.js` as [email address](https://github.com/woocommerce/woocommerce/blob/e35429538776c9984e6a506ca0354792584afcc6/plugins/woocommerce/tests/performance/config.js#L12).
2. There is no any customer with `customer` username on the `wcnightlysmoketest.wpcomstaging.com` test site
3. There is also [customer ID hardcoded in the config](https://github.com/woocommerce/woocommerce/blob/e35429538776c9984e6a506ca0354792584afcc6/plugins/woocommerce/tests/performance/config.js#L15), but this is not related to my code which dynamically extracts the whatever customer ID is used on any test site (so just a sidenote)

How do you think we should proceed with this now? Pinging @tammullen for the `wcnightlysmoketest.wpcomstaging.com` test site, should we maybe create a customer with `customer` username and/or can we change the `customer_username` in the `config.js` file to `customer` and add a new config for the `email`?

Reply:
Hardcoding the customer id would be the fast way and prove to flakiness.
The safe way would be to use the api to search for the customer that has the `customer_username` username and return the id.

Reply:
> The safe way would be to use the api to search for the customer that has the `customer_username` username and return the id.

Which I now see you already implemented. So we're good for the scope of this PR.

> There is no any customer with customer username on the wcnightlysmoketest.wpcomstaging.com test site

Not sure if this site is still in use, maybe for some release runs, but the CI runs are not using it. If it's in use it should be updated.

> There is also [customer ID hardcoded in the config](https://github.com/woocommerce/woocommerce/blob/e35429538776c9984e6a506ca0354792584afcc6/plugins/woocommerce/tests/performance/config.js#L15), but this is not related to my code which dynamically extracts the whatever customer ID is used on any test site (so just a sidenote)

I'm thinking we should change this, and update the customer id via the setup. But we can do this in a follow up PR. 
From the top of my head: the setup should check if the customer exist, if not create it, and return the id which is stored in the config. This way we'll always have the right data in any environment.

Reply:
> ..update the customer id via the setup
I was thinking about the same, so done in a new commit.

Not sure about storing the customer ID in a config.js file, doesn't seem doable with k6 as JS doesn't have permission to write to a config.js file (if that's the intention). We can work on this separately.

Please take another peek and approve if everything's good.
I will wait for @tammullen 's review as well before merging.

Reply:
We should do this in a follow up PR. I'm fine with leaving this as it is in this PR.

To avoid any potential misuse and flakiness I would remove the `customer_user_id` constant from config.js. Form a quick check that constant is only used in `order-filter.js` in `ordersFilter()`

This is what I think we need to do:

- delete the `customer_user_id` from config.js
- extract a function to create the customer if it doesn't exist in the setup module. It should return the user id
- call that function in every test where the customer id is needed and set a global const in there
- update the test functions that required the if to accept the id as an argument

```
// gh-action-pr-requests.js
let customerUserId;

export function setup() {
	setCartCheckoutShortcodes();
	customerUserId = createCustomer();
	addCustomerOrder( customerUserId );
}

...

export function allMerchantFlow() {
	wpLogin();
	homeWCAdmin( { other: false, orders: false, reviews: false, products: false} );
	addOrder();
	orders();
	ordersSearch();
	ordersFilter( customerUserId );
	addProduct();
	products();
	coupons();
}
```

Reply:
That makes sense; please approve the PR so I can merge if @tammullen is okay with the changes.

Reply:
I'll let @tammullen check this out before merging. From my POV consider this approved.
Can you please create an issue for us to track the follow-up work per the above thread?

Reply:
Sure, done in #51536